inc/parser/xhtml.php

```diff
@@ -1108,7 +1108,13 @@ function emaillink($address, $name = null, $returnonly = false) {
     function internalmedia($src, $title = null, $align = null, $width = null,
                            $height = null, $cache = null, $linking = null, $return = false) {
         global $ID;
-        list($src, $hash) = explode('#', $src, 2);
+
+        if (strpos($src, '#') !== false) {
+            list($src, $hash) = explode('#', $src, 2);
+        } else {
+            $hash = null;
+        }
+
```

Comment:
Your timings more or less indicate we're free to choose any of the alternatives as the cost in any given page is going to be insignificant.